src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp

```diff
@@ -260,6 +260,61 @@ std::pair<std::vector<NativeInteger>, uint32_t> ParameterGenerationBGVRNS::compu
     return std::make_pair(moduliQ, totalModSize);
 }
 
+void ParameterGenerationBGVRNS::InitializeFloodingDgg(std::shared_ptr<CryptoParametersBase<DCRTPoly>> cryptoParams,
+                                                      usint numPrimes) const {
+    const auto cryptoParamsBGVRNS = std::static_pointer_cast<CryptoParametersBGVRNS>(cryptoParams);
+
+    KeySwitchTechnique ksTech     = cryptoParamsBGVRNS->GetKeySwitchTechnique();
+    ProxyReEncryptionMode PREMode = cryptoParamsBGVRNS->GetPREMode();
+
+    // compute the flooding distribution parameter based on the security mode for pre
+    // get the re-encryption level and set the level after re-encryption
+    usint ringDimension = cryptoParamsBGVRNS->GetElementParams()->GetRingDimension();
+    double sigma        = cryptoParamsBGVRNS->GetDistributionParameter();
+    double alpha        = cryptoParamsBGVRNS->GetAssuranceMeasure();
+    usint r             = cryptoParamsBGVRNS->GetDigitSize();
+    double log2q        = log2(cryptoParamsBGVRNS->GetElementParams()->GetModulus().ConvertToDouble());
+
+    double B_e       = sqrt(alpha) * sigma;
+    uint32_t auxBits = DCRT_MODULUS::MAX_SIZE;
+    double Bkey      = (cryptoParamsBGVRNS->GetSecretKeyDist() == GAUSSIAN) ? sigma * sqrt(alpha) : 1;
+
+    // get the flooding discrete gaussian distribution
+    auto& dggflooding  = cryptoParamsBGVRNS->GetFloodingDiscreteGaussianGenerator();
+    double noise_param = 1;
+    if (PREMode == FIXED_NOISE_HRA) {
+        noise_param = PRE_SD;
+    }
+    else if (PREMode == NOISE_FLOODING_HRA) {
+        if (ksTech == BV) {
+            if (r > 0) {
+                noise_param = pow(2, STAT_SECURITY_FLOODING) * (1 + 2 * Bkey) * (numPrimes + 1) * (log2q / r + 1) *
+                              sqrt(ringDimension) * (pow(2, r) - 1) * B_e;
+            }
+            else {
+                OPENFHE_THROW(config_error, "Relinwindow value cannot be 0 for BV keyswitching");
+            }
+        }
+        else if (ksTech == HYBRID) {
+            if (r == 0) {
+                double numTowersPerDigit = cryptoParamsBGVRNS->GetNumPerPartQ();
+                int numDigits            = cryptoParamsBGVRNS->GetNumPartQ();
+                noise_param              = numTowersPerDigit * numDigits * sqrt(ringDimension) * B_e * (1 + 2 * Bkey);
+                noise_param += auxBits * (1 + sqrt(ringDimension));
+                noise_param = pow(2, STAT_SECURITY_FLOODING) * noise_param;
+            }
+            else {
+                OPENFHE_THROW(config_error, "Relinwindow value can only  be zero for Hybrid keyswitching");
+            }
+        }
+    }
+    else if (PREMode == DIVIDE_AND_ROUND_HRA) {
+        OPENFHE_THROW(config_error, "Noise Flooding not applicable for PRE DIVIDE_AND_ROUND_HRA mode");
+    }
```

Comment:
INDCPA is the default and it doesn't need this functionality to set the noise flooding parameter. If I add an exception for that option, then it would fail the cases where the premode is not set and stays the default value.